src/D2L.Security.OAuth2/D2L.Security.OAuth2.csproj

```diff
@@ -16,17 +16,26 @@
 		<PackageReference Include="D2L.CodeStyle.Analyzers">
 			<PrivateAssets>All</PrivateAssets>
 		</PackageReference>
+
 		<PackageReference Include="D2L.CodeStyle.Annotations" />
 		<PackageReference Include="D2L.Services.Core.Exceptions" />
-		<PackageReference Include="Newtonsoft.Json" />
+
+		<!-- In .NET Framework use Newtonsoft.Json.
+		     Otherwise, use System.Text.Json. -->
+		<PackageReference Include="Newtonsoft.Json" Condition="'$(TargetFramework)' != 'net50'" />
+
+		<!-- In .NET Framework use System.Runtime.Caching.MemoryCache.
+		     Otherwise, use the NuGet package. -->
+		<Reference Include="System.Runtime.Caching" Condition="'$(TargetFramework)' != 'net50'" />
+		<PackageReference Include="Microsoft.Extensions.Caching.Memory" Condition="'$(TargetFramework)' == 'net50'" />

```

Comment:
nit: consider putting these into conditional `ItemGroup`s instead to clear this up a bit

Reply:
I tried to keep the things grouped by like, alternate package, rather than grouping by target framework 🤷🏼 
I don't really care either way though.

Reply:
Had to do more invasive bifurcation so I went with this.